### PR TITLE
Adjustible animation rate for items and projectiles

### PR DESCRIPTION
**Pull request by islemaster**
Our hazard animation was running too fast, so I'm introducing a way to specify a custom animation rate per item type, in frames per second.

Here you can see the hazard animating much slower than the actor:
slow_animation

### CODE REVIEW

**Comment by islemaster** [comment]
This change points out some really obvious, unfortunate duplication between the `Item` and `Projectile` classes that we can't trivially deduplicate yet because their common base class `Collidable` doesn't follow the same rules.  There's a pretty broken inheritance relationship there.

In a separate PR I'd like to attempt to break this rendering behavior out into something like a `StudioAnimation` that handles framerates and the necessary element/clipRect offsets, so that a client can just say "draw the current frame here."  Then if both `Item` and `Projectile` have a `StudioAnimation` we won't be duplicating functionality.

File: apps/src/studio/Item.js [hunk 1]
```diff
@@ -38,14 +38,26 @@ var Item = function (options) {
...
+Item.prototype.setAnimationRate = function (framesPerSecond) {
+  if (this.animator_) {
+    window.clearInterval(this.animator_);
+  }
   this.animator_ = window.setInterval(function () {
     if (this.loop || this.currentFrame_ + 1 < this.frames) {
       this.currentFrame_ = (this.currentFrame_ + 1) % this.frames;
     }
-  }.bind(this), 50);
+  }.bind(this), 1000 / framesPerSecond);
```
Comment (breville) [inline apps/src/studio/Item.js]: does setInterval cope with non-integer values?

File: apps/src/studio/Item.js [hunk 2]
```diff
@@ -38,14 +38,26 @@ var Item = function (options) {
   this.fadeTime = constants.ITEM_FADE_TIME;
 
   this.currentFrame_ = 0;
+  this.setAnimationRate(utils.valueOr(options.animationRate,
```
Comment (breville) [inline apps/src/studio/Item.js]: might read a bit easier if newline happens before utils, but not sure what our standard is here.

File: apps/src/studio/skins.js [hunk 3]
```diff
@@ -554,7 +554,7 @@ function loadHoc2015x(skin, assetUrl) {
   };
 
   skin.specialItemProperties = {
-    'hazard': { frames: 13, width: 100, height: 100, renderOffset: { x: 0, y: -25}, activity: 'watchActor', speed: constants.SpriteSpeed.VERY_SLOW, isHazard: true }
+    'hazard': { frames: 13, animationRate: 5, width: 100, height: 100, renderOffset: { x: 0, y: -25}, activity: 'watchActor', speed: constants.SpriteSpeed.VERY_SLOW, isHazard: true }
```
Comment (breville) [inline apps/src/studio/skins.js:557]: nice

**Comment by breville** [comment]
looks reasonable to me.

File: apps/src/studio/Item.js (hunk 1 above)
Comment (islemaster) [inline apps/src/studio/Item.js]: Hm, nice catch.  Opinion seems to be that it is safe and generally truncates the value (like `floor`) but that you might get slightly different results on different browsers.  I can certainly make the rounding explicit here.

**Pull request merged**